R/spread_with_duplicates.R

```diff
@@ -0,0 +1,177 @@
+#' Spread key-value pairs across multiple columns.
+#'
+#' @description This function spread one or more key-value pairs across multiple
+#'   columns. If only one value exists per key, then this functions is the same
+#'   as \code{\link[tidyr]{spread}}. Otherwise, aggregation function
+#'   \code{aggfunc} is applied. If no function is passed, then the key-value(s)
+#'   are spread on multiple rows.
+
+#' @param data A dataframe.
+#' @param key,value Column names or positions. This is passed to
+#'   \code{\link[tidyselect]{vars_pull}}. These arguments are passed by
+#'   expression and support \code{\link[rlang]{quasiquotation}} (you can unquote
+#'   column names or column positions)
+#' @param fill If set, missing values will be replaced with this value.
+#' @param convert If \code{TRUE}, \code{\link[utils]{type.convert}} with
+#'   \code{asis = TRUE} will be run on each of the new columns. This is useful
+#'   if the value column was a mix of variables that was coerced to a string. If
+#'   the class of the value column was factor or date, note that will not be
+#'   true of the new columns that are produced, which are coerced to character
+#'   before type conversion.
+#' @param drop If \code{FALSE}, will keep factor levels that don't appear in the
+#'   data, filling in missing combinations with \code{fill}.
+#' @param sep If \code{NULL}, the column names will be taken from the values of
+#'   \code{key} variable. If non-\code{NULL}, the column names will be given by
+#'   "<key_name><sep><key_value>".
+#' @param aggfunc Aggregation function. Default: NA (keep dulicates)
+#'
+#' @return A data.frame.
+#'
+#' @export
+#'
+#' @importFrom purrr map map2 reduce compact
+#' @importFrom magrittr %<>%
+#' @importFrom rlang sym syms is_character
+#' @importFrom dplyr summarize funs rename mutate_all mutate_at filter full_join
+#'   pull %>% one_of group_by vars
+#' @importFrom tidyselect vars_pull enquo
+#' @importFrom tidyr complete_
+#'
+#' @examples
+#' \dontrun{
+#' test1 <- data.frame(
+#' col1 = c(1, 1, 1, 1),
+#' col2 = c("H", "H", "H", "H"),
+#' key = c("A", "B", "C", "C"),
+#' value = c("R", "S", "T", "X"),
+#' stringsAsFactors = FALSE
+#' )
+#' spread_with_duplicates(test1, key, value)
```

Comment:
I get the following error when running this code:

```
Error in vars_pull(names(data), !(!enquo(key))) : 
  could not find function "vars_pull"
```

Reply:
it seems to me your DESCRIPTION file is outdated, I'll add a commit

Reply:
fixed by 51083e1349ebd6b497a670d028614e0d23dd479d

Reply:
thanks!